src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java

```diff
@@ -45,48 +46,53 @@ public ZipFileStatusService(
     public ZipFileStatus getStatusFor(String zipFileName) {
         List<Envelope> envelopes = envelopeRepo.findByZipFileName(zipFileName);
         List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
-        return getZipFileStatus(zipFileName, null, envelopes, events);
+        return getZipFileStatus(zipFileName, null, null, envelopes, events);
+
     }
 
     public List<ZipFileStatus> getStatusByDcn(String documentControlNumber) {
 
         List<String> zipFileNames = scannableItemRepo.findByDcn(documentControlNumber);
         List<ZipFileStatus> zipFileStatusList = new ArrayList<>();
-
-        zipFileNames.stream().forEach(
-            zipFileName ->
+        if (zipFileNames.size() > 0) {
+            zipFileNames.stream().forEach(
+                zipFileName ->
                 zipFileStatusList.add(
                     getZipFileStatus(
                         zipFileName,
                         null,
+                        documentControlNumber,
                         envelopeRepo.findByZipFileName(zipFileName),
                         eventRepo.findByZipFileName(zipFileName)
                     )
                 )
-        );
-
-        return zipFileStatusList;
+            );
+            return zipFileStatusList;
+        }
+        return Arrays.asList(new ZipFileStatus(null, null, documentControlNumber, emptyList(), emptyList()));
     }
 
     public ZipFileStatus getStatusByCcdId(String ccdId) {
         List<Envelope> envelopes = envelopeRepo.findByCcdId(ccdId);
         if (envelopes.size() > 0) {
             String zipFileName = envelopes.get(0).getZipFileName();
             List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
-            return getZipFileStatus(null, ccdId, envelopes, events);
+            return getZipFileStatus(null, ccdId, null, envelopes, events);
         }
-        return getZipFileStatus(null, ccdId, emptyList(), emptyList());
+        return getZipFileStatus(null, ccdId, null, emptyList(), emptyList());
     }
 
     private ZipFileStatus getZipFileStatus(
```

Comment:
you can overload this method instead of adding new param

Reply:
There will be string datatype for parameters ccdid, dcn and filename and return type is always ZipFileStatus. So overloading wouldn't be possible.

Reply:
parameter type will always be string (either dcn, ccdid or filename) and return type will always be zipfilestatus, so overloading wouldn't be possible.

Reply:
meant getZipFileStatusByCcdId or filename ....

Reply:
I can do this but it will do nothing, we will have 3 functions, in the end ZipFileStatus constructor takes all of them as parameters, so we have to pass them as null anyways.